libs/island-ui/core/src/lib/Filter/Filter.tsx

```diff
@@ -38,29 +48,95 @@ export interface FilterProps {
  * like the `isDialog` state with out bloating the childs props.
  */
 interface FilterContextValue {
-  isDialog: boolean
+  variant?: FilterProps['variant']
 }
 
 export const FilterContext = createContext<FilterContextValue>({
-  isDialog: false,
+  variant: undefined,
 })
 
 export const Filter: FC<FilterProps> = ({
+  labelClearAll = '',
   labelClear = '',
   labelOpen = '',
   labelClose = '',
   labelTitle = '',
   labelResult = '',
   resultCount = 0,
-  isDialog = false,
+  align,
+  variant = 'default',
+  filterInput,
   onFilterClear,
   children,
 }) => {
   const dialog = useDialogState()
+  const popover = usePopoverState({
+    placement: 'bottom-start',
+    unstable_flip: true,
+    gutter: 8,
+  })
+
+  const hasFilterInput = !!filterInput
 
   return (
-    <FilterContext.Provider value={{ isDialog }}>
-      {isDialog ? (
+    <FilterContext.Provider value={{ variant }}>
```

Comment:
Could you use the `useContext` hook here, like you do in libs/island-ui/core/src/lib/Filter/FilterMultiChoice/FilterMultiChoice.tsx